jwst/srctype/tests/test_srctype.py

```diff
@@ -151,3 +152,40 @@ def test_tso_types():
     input.meta.visit.tsovisit = True
     output = srctype.set_source_type(input)
     assert output.meta.target.source_type == 'POINT'
+
+def test_nrs_msaspec():
+    """Test for when exposure type is NRS_MSASPEC
+    """
+    input = datamodels.MultiSlitModel()
+    input.meta.exposure.type = "NRS_MSASPEC"
+
+    slits = [{'source_id':1, 'stellarity':0.9},
+             {'source_id':2, 'stellarity':-1},
+             {'source_id':3, 'stellarity':0.5}]
+
+    for slit in slits:
+        input.slits.append(slit)
+
+    result = srctype.set_source_type(input)
+
+    assert(result.slits[0].source_type == 'POINT')
+    assert(result.slits[1].source_type == 'UNKNOWN')
+    assert(result.slits[2].source_type == 'EXTENDED')
+
+def test_is_tso():
+    """ Test for when visit is tso.
+    """
+    input = datamodels.ImageModel((10,10))
+    input.meta.exposure.type = "FGS_DARK"
```

Comment:
Should probably use a TSO exptype here.  In fact, there's a `test_tso_types()` test above.  Maybe parametrize that one and test all the TSO types?

Reply:
Something like
```
@pytest.mark.parametrize("exptype", ["NRS_BRIGHTOBJ", "NRC_TSGRISM", "NIS_SOSS",
    "MIR_LRS-SLITLESS"])
def test_tso_types(exptype):
    """Check that TSO exptypes get set to POINT"""
    input = datamodels.ImageModel()
    input.meta.observation.bkgdtarg = False
    input.meta.dither.primary_type = 'NONE'
    input.meta.visit.tsovisit = True
    input.meta.exposure.type = exptype

    result = srctype.set_source_type(input)

    assert result.meta.target.source_type == 'POINT'
```